src/ErrorHandling/ErrorModal.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { Modal } from "../ui/React/Modal";
+import { Box, Button, Typography } from "@mui/material/";
+import { errorModalsAreSuppressed, type ErrorRecord, ErrorState, toggleSuppressErrorModals } from "./ErrorState";
+import { Router } from "../ui/GameRoot";
+import { SimplePage, ToastVariant } from "@enums";
+import { useRerender } from "../ui/React/hooks";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+import { LogBoxEvents } from "../ui/React/LogBoxManager";
+import { recentScripts } from "../Netscript/RecentScripts";
+import { SnackbarEvents } from "../ui/React/Snackbar";
+import { Settings } from "../Settings/Settings";
+
+export function ErrorModal(): React.ReactElement {
+  const { classes } = useStyles();
+  const rerender = useRerender();
+  const [error, setError] = useState<ErrorRecord | null>(ErrorState.ActiveError);
+
+  useEffect(() => {
+    const listener = (newError: ErrorRecord) => {
+      if (newError.force || (Router.page() !== SimplePage.RecentErrors && !errorModalsAreSuppressed())) {
+        setError(newError);
+        rerender();
+      }
```

Comment:
```suggestion
      if (newError.force || (Router.page() !== SimplePage.RecentErrors && !errorModalsAreSuppressed())) {
        setError(newError);
        rerender();
      } else {
        ErrorState.ActiveError = null;
        setError(null);
      }
```
Bug:
- Run the test script. Close error popups.
- Switch to the "Active Scripts" tab.
- Switch to the "Active" tab, then back to the "Recent Errors" tab. (This step is a workaround for another bug).
- Run the test script again via the log modal. No modal is shown.
- Switch to another tab. Run the test script again via the log modal. No modal is shown.

This is the same bug as in the last review. With the current design, you must be extremely careful with `ErrorState.ActiveError`. If you forget to reset it, the error popup will be mistakenly suppressed.

Edit: Please test this bug again after you fix all problems. Maybe I also forgot something or the fix here is flawed.